File: AtCoder/A_Online_Shopping.go
package main

import "fmt"

func main() {
	var N, S, K int
	fmt.Scan(&N, &S, &K)

	var ans int
	for i := 0; i < N; i++ {
		var p, q int
		fmt.Scan(&p, &q)
		ans += p * q
	}
	if ans < S {
		ans += K
	}

	fmt.Println(ans)
}

File: AtCoder/A_Online_Shopping.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        s: i32,
        k: i32,
        pq: [(i32, i32); n], 
    }

    let mut ans: i32 = 0;
    for (p, q) in pq {
        ans += p * q;
    }

    ans += if ans >= s { 0 } else { k };
    println!("{}", ans)
}